--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -713,11 +713,14 @@ config("compiler") {
       }
     }
 
-    # TODO(crbug.com/40192287): Investigate why/if this should be needed.
-    if (is_win) {
-      cflags += [ "/clang:-ffp-contract=off" ]
-    } else {
-      cflags += [ "-ffp-contract=off" ]
+    # Enable fast FP contract for AVX2/AVX512 builds (Chromium_Clang/Thorium)
+    if (use_avx2 || use_avx512 || use_fma) {
+      if (is_win) {
+        cflags += [ "/clang:-ffp-contract=fast" ]
+      } else {
+        cflags += [ "-ffp-contract=fast" ]
+      }
     }
 
     # Enable ELF CREL (see crbug.com/357878242) for all platforms that use ELF.
